metrics/prometheus.go

```diff
@@ -2,21 +2,45 @@ package metrics
 
 import (
 	"flag"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"log"
 	"net/http"
 	"strconv"
 )
 
 type PrometheusConfig struct {
+	//metricPort to which the metrics would be sent
 	metricPort int
+
+	//metric collector
+	offsetMetrics *prometheus.GaugeVec
 }
```

Comment:
let's ensure we maintain separation of concern, we don't need config object to store metric objects too and later using the same object to ingest either. for prometheus usecase, we can simply define stateless implementation of registry interface to start prometheus server and ingest metrics as required

```suggestion
var (
	// initialize all the metrics here
	offsetMetrics = prometheus.NewGaugeVec(
		prometheus.GaugeOpts{
			Name: "offset_metrics",
			Help: "The metric represent all offset related metrics for dmux",
		}, []string{"key"})
)

type PrometheusConfig struct {
	//metricPort to which the metrics would be sent
	metricPort int
}

type PrometheusRegistry struct {
	// stateless
}
```